test/smoke/smoke_test.js

```diff
@@ -0,0 +1,102 @@
+const path = require('path');
+const childProcess = require('child_process');
+const util = require('util');
+const fs = require('fs');
+
+// Catch errors from failing promises
+process.on('unhandledRejection', function (error) {
+  console.error(error && error.stack);
+  process.exit(3);
+});
+
+const configsDir = path.normalize(path.join(__dirname, '../configs'));
+
+const processEnv = Object.assign({}, process.env);
+const ciMode = process.argv.includes('ci');
+if (ciMode) {
+  processEnv.BACKSTOP_CI_MODE = '1';
+}
+
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function runSmokeTest (description, commands) {
+  const child = childProcess.spawn(commands[0], commands.slice(1),
+    { cwd: configsDir, env: processEnv }
+  );
+
+  const requiredStdout = [
```

Comment:
This sort of thing should be placed at the top of this script with explicit comments around how to use this array to handle different test scenarios.

In this case we should name it something very clear.